speechbrain/inference/ASR.py

```diff
@@ -397,43 +398,251 @@ class WhisperASR(Pretrained):
+        language_probs : List[Dict[str, float]], length = n_audio
+            list of dictionaries containing the probability distribution over all languages.
+        """
+        languages = [self.mods.whisper.language] * mel.shape[0]
+        lang_probs = None
+
+        if self.mods.whisper.language is None or task == "lang_id":
+            lang_tokens, lang_probs = self.mods.whisper.detect_language(mel)
+            languages = [max(probs, key=probs.get) for probs in lang_probs]
+            self.mods.decoder.set_lang_tokens(lang_tokens)
+        return languages, lang_probs
+
+    @torch.no_grad()
+    def transcribe_file(
+        self,
+        input: Union[str, torch.Tensor],
+        task: Optional[str] = None,
+        initial_prompt: Optional[str] = None,
+        logprob_threshold: Optional[float] = -1.0,
+        no_speech_threshold=0.6,
+        condition_on_previous_text: bool = False,
+        verbose: bool = False,
+    ):
+        """Transcribes the given audiofile into a sequence of words.
+
+        Arguments
+        ---------
+        input : Union[str, torch.Tensor]
+            Path to audio file which to transcribe or torch tensor of audio.
+        task : Optional[str]
+            The task to perform. If None, the default task is the one passed in the Whisper model.
+        initial_prompt : Optional[str]
+            The initial prompt to condition the model on.
+        logprob_threshold : Optional[float]
+            The log probability threshold to continue decoding the current segment.
+        no_speech_threshold : float
+            The threshold to skip decoding segment if the no_speech_prob is higher than this value.
+        condition_on_previous_text : bool
+            If True, the model will be condition on the last 224 tokens.
+        verbose : bool
+            If True, print the transcription of each segment.
+
         Returns
         -------
         str
             The audiofile transcription produced by this ASR system.
+        results
+            The results containing transcription, tokens, prompt, avg_log_probs, and no_speech_prob for each segment.
         """
-        waveform = self.load_audio(path)
-        # Fake a batch:
-        batch = waveform.unsqueeze(0)
+        if task is not None:
+            if task in self.TASKS:
+                if task != "lang_id":
+                    self.mods.decoder.set_task(task)
+            else:
+                raise ValueError(
+                    f"Task {task} not supported. Supported tasks are {self.TASKS}"
+                )
+
+        if isinstance(input, str):
+            audio = self.load_audio(input).float().to(self.device).unsqueeze(0)
+        else:
+            audio = input.float().to(self.device)
+
+        # create chunks of 30 seconds, if the audio is longer than 30 seconds
+        N_FRAMES = 30 * self.hparams.sample_rate
+        segments = split_fixed_chunks(audio, N_FRAMES, dim=-1)
```

Comment:
Maybe we could extract `StreamingASR._get_audio_stream` and this snippet:

```python
        if use_torchaudio_streaming:
            streamer = torchaudio.io.StreamReader(path)
            chunks = self._get_audio_stream(streamer, chunk_size)
        else:
            waveform = self.load_audio(path, **kwargs)
            batch = waveform.unsqueeze(0)  # create batch dim
            chunks = split_fixed_chunks(batch, chunk_size)
```

It would let you easily bring the same stream handling logic as `StreamingASR` does.

Reply:
done.